### PR TITLE
test(e2e): fix masthead overflow e2e tests

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

No related issue

### Description

More of a bandaid fix, this adds a wait for the Masthead overflow scroll to complete before checking the element is disabled.

### Changelog

**Changed**

- `masthead.e2e.js`

### CODE REVIEW

**Review by ariellalgilmore — COMMENTED** [review]
lol thank you! i could have added this. just was trying to see is something else worked. Should we increase the time for the L1 too?

**Comment by jeffchew** [comment]
As discussed, switched up the tests to do proper waiting of classes before executing the `expect` commands.

**Review by ariellalgilmore — APPROVED** [review]
LGTM! Thank you for the updates!

File: packages/react/tests/e2e-storybook/cypress/integration/Masthead/Masthead.e2e.js [hunk 1]
```diff
@@ -292,27 +292,12 @@ describe('Masthead | custom (desktop)', () => {
...
-      </style>
-    `
-      );
-    });
-
     cy.get('.bx--header__nav-caret-right').click();
 
-    cy.get('.bx--header__nav-caret-right-container').then($button => {
-      expect($button).to.have.attr('hidden');
+    cy.get('.bx--header__nav-caret-right-container').then(() => {
```
Comment (jeffchew) [inline packages/react/tests/e2e-storybook/cypress/integration/Masthead/Masthead.e2e.js:297]: Realize that this isn't checking for the hidden attribute before continuing. Will try to fix that today.

**Pull request merged**